prow/README.md

```diff
@@ -4,206 +4,383 @@ Metal3 Prow dashboard: <https://prow.apps.test.metal3.io>
+   - `repo` scope for full control of private repositories
+   - `workflow`
+   - `admin:org_hook` for a github org
 
-   ```shell
-   openssl rand -hex 20 > /path/hmac-token
-   kubectl create secret generic hmac-token --from-file=hmac=/path/hmac-token
-   ```
+   The token will be referred to as `${GITHUB_TOKEN}`.
 
-1. Create a personal access token for the GitHub bot account. This should be
-   done from [metal3-io-bot](https://github.com/metal3-io-bot) GitHub bot
-   account. You can follow this [link](https://github.com/settings/tokens)
-   to create the token. When generating the token, make sure you have only the
-   following scopes checked in.
+1. Create a personal access token for the cherry-picker bot.
+   This should be done from the [metal3-cherrypick-bot](https://github.com/metal3-cherrypick-bot)
+   GitHub bot account.
+   You can follow this [link](https://github.com/settings/tokens) to create the
+   token. When generating the token, make sure you have only the following scopes
+   checked in.
 
-   * `repo` scope for full control of private repositories
-   * `admin:org_hook` for a github org
+   - `repo`
+   - `workflow`
+   - `admin:org/read:org` (not all of admin:org, just the sub-item read:org)
 
-   ![token-scopes](images/token-scopes.png)
+   The token will be referred to as `${CHERRYPICK_TOKEN}`.
 
-   Create a secret out of that token:
+1. Create a GitHub webhook for <https://prow.apps.test.metal3.io/hook> using the
+   HMAC token generated earlier.
+   Add the URL and token as below. Select **"Send me everything"**, and for
+   Content type: **application/json**.
 
-   ```shell
-   kubectl create secret generic github-token \
-       --from-file=token=/path/github-token
+   ![webhook configuration](images/webhook.png)
+
+## Generate secret files
+
+Files with credentials and other sensitive information are not stored in this repository.
+You will need to add them manually before you can apply any manifests and build
+the kustomizations.
+CAPO needs access to the OpenStack API and so does the external cloud-provider.
+Prow needs a GitHub token for accessing GitHub, a HMAC token for validating webhook
+requests, and S3 credentials for storing logs and similar.
+
+If you are deploying from scratch or rotating credentials, please make sure to
+save them in a secure place after creating them.
+If there is an existing instance, you most likely just have to take the credentials
+from the secure place and generate the files below from them.
+
+Please set the following environment variables with the relevant credentials.
+Then you will be able to just copy and paste the snippets below.
+
+- `APP_CRED_ID`
+- `APP_CRED_SECRET`
+- `S3_ACCESS_KEY`
+- `S3_SECRET_KEY`
+- `HMAC_TOKEN`
+- `GITHUB_TOKEN`
+- `CHERRYPICK_TOKEN`
+
+Now you are ready to create the files.
+
+1. Create `clouds.yaml` secret used by the CAPO controllers to manage the infrastructure.
+
+   ```bash
+   cat > capo-cluster/clouds.yaml <<-EOF
```

Comment:
`-EOF` is removing leading tabs, which you don't use here, so remove the `-`, it does nothing here. It would be wrong to remove leading spaces anyways.